backend/battle/tests/test_view.py

```diff
@@ -285,3 +287,136 @@ def test_if_returns_error_when_battle_does_not_exist(self):
+    def setUp(self):
+        super().setUp()
+        self.battle = baker.make('battle.Battle', creator=self.user)
+        names = ['pikachu', 'bulbasaur', 'pidgeot', 'wrong_name']
+        for pokemon_name in names:
+            baker.make("pokemon.Pokemon", name=pokemon_name, attack=60, defense=15, hp=10)
+
+    def test_form_is_valid(self):
+        pokemons_data = {
+            "battle": self.battle.id,
+            "trainer": self.user.id,
+            "pokemon_1": 'pikachu',
+            "pokemon_2": 'bulbasaur',
+            "pokemon_3": 'pidgeot',
+            "position_pkn_1": 1,
+            "position_pkn_2": 2,
+            "position_pkn_3": 3,
+        }
+        form = TeamForm(data=pokemons_data)
+        self.assertTrue(form.is_valid())
+
+    def test_form_saved_successfully(self):
+        pokemons_data = {
+            "battle": self.battle.id,
+            "trainer": self.user.id,
+            "pokemon_1": 'pikachu',
+            "pokemon_2": 'bulbasaur',
+            "pokemon_3": 'pidgeot',
+            "position_pkn_1": 1,
+            "position_pkn_2": 2,
+            "position_pkn_3": 3,
+        }
+        form = TeamForm(data=pokemons_data)
+        self.assertTrue(form.is_valid())
+        instance = form.save()
+        self.assertEqual(Team.objects.get(battle=self.battle, trainer=self.user), instance)
+
+    def test_form_returns_error_pokemons_missing(self):
+        pokemons_data = {
+            "battle": self.battle.id,
+            "trainer": self.user.id,
+            "position_pkn_1": 1,
+            "position_pkn_2": 2,
+            "position_pkn_3": 3,
+        }
+        form = TeamForm(data=pokemons_data)
+        self.assertFalse(form.is_valid())
+        with self.assertRaisesMessage(ValidationError, 'ERROR: Select all pokemons'):
+            form.save()
+
+    def test_form_returns_error_positions_missing(self):
+        pokemons_data = {
+            "battle": self.battle.id,
+            "trainer": self.user.id,
+            "pokemon_1": 'pikachu',
+            "pokemon_2": 'bulbasaur',
+            "pokemon_3": 'pidgeot',
+        }
+        form = TeamForm(data=pokemons_data)
+        self.assertFalse(form.is_valid())
+        with self.assertRaisesMessage(ValidationError, 'ERROR: Select all positions'):
+            form.save()
+
+    def test_form_returns_error_with_battle_invalid(self):
+        pokemons_data = {
+            "battle": 100,
+            "trainer": self.user.id,
+            "pokemon_1": 'pikachu',
+            "pokemon_2": 'bulbasaur',
+            "pokemon_3": 'pidgeot',
+            "position_pkn_1": 1,
+            "position_pkn_2": 2,
+            "position_pkn_3": 3,
+        }
+        form = TeamForm(data=pokemons_data)
+        self.assertFalse(form.is_valid())
+        with self.assertRaisesMessage(ValidationError, 'ERROR: Select a valid battle'):
+            form.save()
+
+    def test_form_returns_error_with_trainer_missing(self):
```

Comment:
As I mentioned earlier lets assume that when a battle exists the trainer also exists

Reply:
Okay :)